### PR TITLE
[core] #502 Extracts packages from source files

### LINKED ISSUE #502: Extract package name from source file instead of file name
Currently, we extract package name from file name based on some assumptions which are not true always. A more robust approach would be to extract packages from source files.

### PR DESCRIPTION
**Pull request by keerath**
fixes #502
closes #505

### CODE REVIEW

**Comment by keerath** [comment]
@amit-shar PTAL.!!!

File: core/src/main/scala/com/kodebeagle/crawler/ZipBasicParser.scala [hunk 1]
```diff
@@ -51,42 +51,59 @@ object ZipBasicParser extends Logger {
...
+    (javaFileList.toList, scalaFileList.toList, extractPackages(fileContentList),
+      toRepository(repoFileNameInfo, statistics))
+  }
+
+  private def extractPackages(fileContentList: List[String]) = {
+    val allPackages = mutable.Set[String]()
+    import scala.io.Source
+    fileContentList.foreach { fileContent =>
+      val lines = Source.fromString(fileContent).getLines()
+      val PACKAGE = "package"
```
Comment (amit-shar) [inline core/src/main/scala/com/kodebeagle/crawler/ZipBasicParser.scala]: use "package  " instead.

File: core/src/main/scala/com/kodebeagle/crawler/ZipBasicParser.scala [hunk 2]
```diff
@@ -51,42 +51,59 @@ object ZipBasicParser extends Logger {
...
+      lines.filter(_.trim.startsWith(PACKAGE)).foreach { line =>
+        val strippedLine = line.stripPrefix(PACKAGE).trim
```
Comment (amit-shar) [inline core/src/main/scala/com/kodebeagle/crawler/ZipBasicParser.scala]: check for first occurrence of 'package' only.

File: core/src/main/scala/com/kodebeagle/crawler/ZipBasicParser.scala [hunk 3]
```diff
@@ -51,42 +51,59 @@ object ZipBasicParser extends Logger {
...
+        val indexOfSemiColon = strippedLine.indexOf(";")
+        if (indexOfSemiColon == -1) {
+          allPackages += strippedLine
```
Comment (amit-shar) [inline core/src/main/scala/com/kodebeagle/crawler/ZipBasicParser.scala]: write comments for java/scala packages.

**Comment by amit-shar** [comment]
LGTM!!

**Pull request merged**